src/dht_proxy_server.cpp

```diff
@@ -352,24 +352,26 @@ DhtProxyServer::subscribe(const std::shared_ptr<restbed::Session>& session)
                     // Check if listener is already present and refresh timeout if launched
 
                     auto pushListener = pushListeners_.emplace(pushToken, PushListener{}).first;
-                    auto listeners = pushListener->second.listeners.emplace(infoHash, std::vector<Listener>{}).first;
-                    for (auto& listener: listeners->second) {
-                        if (listener.token == tokenFromReq) {
-                            {
-                                std::lock_guard<std::mutex> l(schedulerLock_);
-                                scheduler_.edit(listener.expireJob, scheduler_.time() + OP_TIMEOUT);
-                            }
-                            s->close(restbed::OK, "{\"token\": " + std::to_string(listener.token) + "}\n");
-                            schedulerCv_.notify_one();
-                            return;
+                    auto list = pushListener->second.listeners.emplace(infoHash, Listener{});
+                    auto& listener = list.first->second;
+                    if (not list.second) {
+                        // Refresh existing listener
+                        listener.token = tokenFromReq;
```

Comment:
Warning, when we refresh this token, it replaces the previous one. But the previous one is still used in the push notifications because of the comment below.